consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/PostMergeContext.java

```diff
@@ -135,7 +137,9 @@ public boolean isSyncing() {
     return Optional.ofNullable(syncState.get()).map(s -> !s.isInSync()).orElse(Boolean.TRUE)
         // this is necessary for when we do not have a sync target yet, like at startup.
         // not being stopped at ttd implies we are syncing.
-        && !syncState.get().hasReachedTerminalDifficulty().orElse(Boolean.FALSE);
+        && Optional.ofNullable(syncState.get())
+            .map(s -> !(s.hasReachedTerminalDifficulty().orElse(Boolean.FALSE)))
+            .orElse(Boolean.FALSE);
```

Comment:
There's quite a few branches of logic in this isSyncing method, are they all covered by unit tests?

I _think_ the final orElse should actually be TRUE.

If the first statement on L137 is saying "when syncState is null, then return TRUE" then I think this statement behave the same

Reply:
+1

Reply:
fixed